matchzoo/engine/base_preprocessor.py

```diff
@@ -119,34 +119,36 @@ def segmentation(self, inputs: list, stage: str) -> datapack.DataPack:
         # prepare data pack.
         inputs = pd.DataFrame(inputs, columns=columns_all)
         # get mapping columns (idx left and idx right)
-        mapping = inputs[columns_mapping]
-        # Convert mapping to list of records.
-        mapping = mapping.set_index('id_left').to_dict(orient='index')
+        data = inputs[columns_data]
 
         # Segment input into 2 dataframes.
-        data_left = inputs[['id_left', 'text_left']
-                           ].drop_duplicates(['id_left'])
-        data_left.columns = columns_data
+        mapping_left = inputs[['id_left', 'text_left']].drop_duplicates(
+            ['id_left']
+        )
+        mapping_left.columns = columns_mapping
 
-        data_right = inputs[['id_right', 'text_right']
-                            ].drop_duplicates(['id_right'])
-        data_right.columns = columns_data
+        mapping_right = inputs[['id_right', 'text_right']].drop_duplicates(
+            ['id_right']
+        )
+        mapping_right.columns = columns_mapping
 
-        data = pd.concat([data_left, data_right])
+        mapping = pd.concat([mapping_left, mapping_right])
+
+        mapping = mapping.set_index('id').to_dict(orient='index')
 
         return datapack.DataPack(data=data,
```

Comment:
same, now data is dataframe instead of list, it will be converted to dataframe aginn in datapack `__init__.py`